docroot/modules/humsci/hs_admin/hs_admin.module

```diff
@@ -19,6 +20,9 @@ function hs_admin_page_attachments(&$variables) {
   if ($logged_in) {
     $variables['#attached']['library'][] = 'hs_admin/hs_admin';
   }
+
+  // Show shortcuts toolbar when user is logged in.
+  $variables['#attached']['library'][] = 'hs_admin/default_shortcuts';
```

Comment:
I still think you should add the access check for `access toolbar`. That'll avoid the JS adding to anon users. Or you could just add it to the `hs_admin` library.

Reply:
We need the script to run for anon users to clear the `Drupal.toolbar.activeTabID` localstorage item when the users log out. Otherwise that item will persist and we wouldn't be able to switch the selected toolbar only on login.

Reply:
I see now. I'm not a big fan of that approach since it runs on every page for ever user and every ajax. I feel there's probably a way to accomplish this but I'm unable to think of an alternative at this time. The speed at which those couple lines runs is fast and doesn't impact anywhere else on the site, so that's the main thing.

Any chance you've search d.o to see if someone mentioned anything about the storage sticking around?

Reply:
> I'm not a big fan of that approach since it runs on every page for ever user and every ajax.
Yes, I agree. That's why we did the initial implementation using state, but I understand why that approach is not ideal either.

Even when the JS runs in every case, we think that it's impact on performance is small, specially for anon users. If you consider that's helpful, we can add extra checks to return early if the code is executed in an ajax call (using `once` or checking that `context` matches whole document).

> Any chance you've search d.o to see if someone mentioned anything about the storage sticking around?
No, we couldn't find anything relevant for this use case. From what I see, persisting the selected toolbar state is the designed behavior, but we need to override it to improve usability for our users.

Reply:
> we can add extra checks to return early if the code is executed in an ajax call

I actually think doing that would decrease the performance of this part, so I think what you currently have is best case for this situation. It all runs synchronously with other behaviors I believe anyways.

Thanks for checking d.o. That's too bad it's kinda designed that way. It would be nice if they provided a setting to change that behavior. :shrug: